prelude/prelude.purs

```diff
@@ -570,16 +687,27 @@ module Prelude
 
   infixr 5 ++
 
+  -- | Alias for (<>), typically used for String or Array concatenation as a
+  -- | style choice.
+  -- |
+  -- | `[1, 2, 3] ++ [4, 5] == [1, 2, 3, 4, 5]`
   (++) :: forall s. (Semigroup s) => s -> s -> s
   (++) = (<>)
 
 module Data.Function where
 
+  -- | Apply a function to two arguments, then combine the results with a second
+  -- | function.
+  -- |
+  -- | `on (*) length "argument1" "arg2" == 9 * 4`
```

Comment:
I always thought this function was intended to be used infix and point-free - maybe something like this might be a bit more natural?
```
let f = compare `on` length
in f "argument1" "arg2"
```